Rubberduck.CodeAnalysis/Inspections/Concrete/AssignmentNotUsedInspection.cs

```diff
@@ -108,6 +111,140 @@ private static bool IsAssignmentOfNothing(IdentifierReference reference)
                 && setStmtContext.expression().GetText().Equals(Tokens.Nothing);
         }
 
+        /// <summary>
+        /// Filters false positive result references due to GoTo and Resume statements.  e.g., 
+        /// An ErrorHandler block that branches execution to a location where the asignment may be used. 
+        /// </summary>
+        /// <remarks>
+        /// Excludes Assignment references that meet the following conditions:
+        /// 1. Preceed a GoTo or Resume statement that branches execution to a line before the 
+        ///     assignment reference, and
+        /// 2. A non-assignment reference is present on a line that is:
+        ///     a) At or below the start of the execution branch, and 
+        ///     b) Above the next ExitStatement line (if one exists) or the end of the procedure
+        /// </remarks>
+        /// <param name="resultCandidate"></param>
+        /// <param name="finder"></param>
+        /// <returns></returns>
+        private static bool IsPotentiallyUsedViaResumeOrGoToExecutionBranch(IdentifierReference resultCandidate, DeclarationFinder finder)
+        {
+            if (!resultCandidate.Declaration.References.Any(rf => !rf.IsAssignment)) { return false; }
+
+            var labelIdLineNumberPairs = finder.DeclarationsWithType(DeclarationType.LineLabel)
+                                                .Where(label => resultCandidate.ParentScoping.Equals(label.ParentDeclaration))
+                                                .Select(lbl => (lbl.IdentifierName, lbl.Context.Start.Line));
+
+            return GotoExecutionBranchPotentiallyUsesVariable(resultCandidate, labelIdLineNumberPairs) 
+                || ResumeExecutionBranchPotentiallyUsesVariable(resultCandidate, labelIdLineNumberPairs);
+        }
+
+        private static bool GotoExecutionBranchPotentiallyUsesVariable(IdentifierReference resultCandidate, IEnumerable<(string, int)> labelIdLineNumberPairs)
+        {
+            var gotoCtxts = resultCandidate.ParentScoping.Context.GetDescendents<VBAParser.GoToStmtContext>()
+                .Where(gotoCtxt => gotoCtxt.Start.Line > resultCandidate.Context.Start.Line);
+
+            if (!gotoCtxts.Any()) { return false; }
+
+            var gotoStmt = GetFirstContextAfterLine(gotoCtxts, resultCandidate.Context.Start.Line);
+
+            if (gotoStmt == null) { return false; }
```

Comment:
This could possibly be reformulated using a `Sort` and `FirstOrDefault` something like the following air-code

```
var gotoStmt = resultCandidate.ParentScoping.Context.GetDescendents<VBAParser.GoToStmtContext>()
  .Where(gotoCtxt => {
     var gotoStart = gotoCtxt.Start; var resultStart = resultCandidate.Context.Start;
     return gotoStart.Line > resultStart.Line || gotoStart.Line == resultStart.Line && gotoStart.Column > resultStart.Column
  })
  .OrderBy(ctx => ctx.Start.Line)
  .FirstOrDefault();
if (gotoStmt == default) { return false; }
```

But I'd understand if in the service of readability this were abandoned.

Note that I expanded the goto context check to account for same-line GoTo's to account for that edge-case.

Reply:
Handling the same-line `GoTo` and `Resume` scenarios drove a fair amount of changes - all for the better.

TBH I have never used or come across the use of the ":" to place multiple statements on the same line (...kind of unbelievable).  Your initial comment/example had me a confused for a little while.  Thank you for spotting that missing use case.